### ISSUE TITLE
Some question about code.

### ISSUE THREAD

**Issue by Ayanami2019** [voidstrike/TDPNet#2]
Thank you for sharing the code, I want to know how to train on all categories, or is it necessary to train one by one.

**Comment by voidstrike** [comment]
Hey, @Ayanami2019, thanks for your interest!

We do not provide the multi-category training code due to our poor version control, and we currently have no plan to re-write the code. Sorry for the inconvenience.

If you are willing to modify the code, here are some steps I remember.

1. Train Phase 1 as usual (or use a pretrained PointNet encoder), compute the KMean centroids for each class -> This will give you a `(# of classes, # of prototypes, 1024)` tensor.
2. Transpose this tensor with axis `(1, 0, 2)`, replace `proto_feat` in `Proto2Dto3DCluster` with the sub-tensor. Let's say you have 10 classes and 8 prototypes for each. There are 8 `Proto2Dto3DCluster` and each of them should now have a `proto_feat` with shape `(10, 1024)`
3. Add label information to the forward function and modify the training code -- the label information is already provided [here](https://github.com/voidstrike/TDPNet/blob/b75a04a824421de9b4cc290f7222fe94b398fb78/src/datasets/mv_dataset.py#L133)
4. Pay attention to [this line](https://github.com/voidstrike/TDPNet/blob/b75a04a824421de9b4cc290f7222fe94b398fb78/src/model/net_components.py#L52). Please notice that we `repeat` the `proto_feat` K times (K=batch_size). Instead of using `repeat`, you should modify this line to build `batch_proto` based on the given label. -- Recall what we did in step 2.

Again, really sorry for the inconvenience. Last, maybe you can just train the multi-category model as a single-category model -- In our later research, we found that this might provide even better performance.

**Comment by voidstrike** [comment]
Mis-click

**Comment by Ayanami2019** [comment]
Thanks for your answer!
